map2loop/utils.py

```diff
@@ -0,0 +1,104 @@
+import numpy
+from shapely.geometry import Point
+import beartype
+
+
+@beartype.beartype
+def setup_grid(bounding_box: dict):
+    """
+    Setup the grid for interpolation
+
+    Args:
+        bounding_box
+
+    Returns:
+        xi, yi (numpy.ndarray, numpy.ndarray): The x and y coordinates of the grid points.
+    """
+    # Define the desired cell size
+    cell_size = 0.01 * (bounding_box["maxx"] - bounding_box["minx"])
+
+    # Calculate the grid resolution
+    grid_resolution = round((bounding_box["maxx"] - bounding_box["minx"]) / cell_size)
+
+    # Generate the grid
+    x = numpy.linspace(bounding_box["minx"], bounding_box["maxx"], grid_resolution)
+    y = numpy.linspace(bounding_box["miny"], bounding_box["maxy"], grid_resolution)
+    xi, yi = numpy.meshgrid(x, y)
+    xi = xi.flatten()
+    yi = yi.flatten()
+
+    return xi, yi
+
+
+def strike_dip_vector(strike, dip):
+    """
+    This function calculates the strike-dip vector from the given strike and dip angles.
+
+    Parameters:
+    strike (float or array-like): The strike angle(s) in degrees. Can be a single value or an array of values.
+    dip (float or array-like): The dip angle(s) in degrees. Can be a single value or an array of values.
+
+    Returns:
+    vec (numpy.ndarray): The calculated strike-dip vector(s). Each row corresponds to a vector,
+    and the columns correspond to the x, y, and z components of the vector.
+
+    Note:
+    This code is adapted from LoopStructural.
+    """
+
+    # Initialize a zero vector with the same length as the input strike and dip angles
+    vec = numpy.zeros((len(strike), 3))
+
+    # Convert the strike and dip angles from degrees to radians
+    s_r = numpy.deg2rad(strike)
+    d_r = numpy.deg2rad(dip)
+
+    # Calculate the x, y, and z components of the strike-dip vector
+    vec[:, 0] = numpy.sin(d_r) * numpy.cos(s_r)
+    vec[:, 1] = -numpy.sin(d_r) * numpy.sin(s_r)
+    vec[:, 2] = numpy.cos(d_r)
+
+    # Normalize the strike-dip vector
+    vec /= numpy.linalg.norm(vec, axis=1)[:, None]
+
+    return vec
+
+
+def normal_vector_to_dipdirection_dip(nx, ny, nz):
```

Comment:
I prefer to keep the vectors in a Nx3 array rather than specifying the components as individual arrays. I think it makes the code neater.

Reply:
I agree about making a loop-core library because there is an overlap with foldoptlib as well. We can discuss it in one of the next meetings.

Reply:
using only Nx3 array instead of components